clients/src/main/java/org/apache/kafka/common/network/SaslChannelBuilder.java

```diff
@@ -52,7 +55,15 @@ public SaslChannelBuilder(Mode mode, LoginType loginType, SecurityProtocol secur
     public void configure(Map<String, ?> configs) throws KafkaException {
         try {
             this.configs = configs;
-            this.loginManager = LoginManager.acquireLoginManager(loginType, configs);
+            String mechanism = (String) this.configs.get(SaslConfigs.SASL_MECHANISM);
```

Comment:
Could we just use configs.getString and avoid casting? There are a few other places like that.